app/src/main/java/com/example/grutor/Adapters/SubjectAdapter.java

```diff
@@ -43,41 +48,94 @@ public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
     @Override
     public void onBindViewHolder(@NonNull SubjectAdapter.ViewHolder holder, @SuppressLint("RecyclerView") int position) {
         holder.title.setText(titles.get(position));
-        holder.gridIcon.setImageResource(images.get(position));
+
+        set4KCards(holder);
+
         holder.itemView.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                // Display a toast message to the user
-                // Display the title of the card
-                Toast.makeText(inflater.getContext(), "You clicked: " + titles.get(position), Toast.LENGTH_SHORT).show();
                 getTutoringHelp(position);
             }
 
             private void getTutoringHelp(int index) {
                 Bundle bundle = new Bundle();
                 bundle.putString("subject", titles.get(position));
                 Intent i = new Intent(context, DetailActivity.class);
-                // only way to access my bundled String
                 i.putExtras(bundle);
                 context.startActivity(i);
             }
         });
     }
 
+    private void set4KCards(ViewHolder holder) {
+        if (holder.title.getText().toString().equals("Math")) {
+            String MATH_CARD_URL = "https://p2.piqsels.com/preview/939/797/86/calculator-the-calculation-of-casio-zero.jpg";
+            Glide.with(context).load(MATH_CARD_URL)
+                    .into(new SimpleTarget<Drawable>() {
+                        @Override
+                        public void onResourceReady(@NonNull Drawable resource, @Nullable com.bumptech.glide.request.transition.Transition<? super Drawable> transition) {
+                            holder.rlCardImageContainer.setBackground(resource);
+                        }});
```

Comment:
for the static icon on the Ui, it would better put the icon into local resource.